AirLib/include/common/AirSimSettings.hpp

```
@@ -1040,6 +1083,129 @@ struct AirSimSettings {
+
+        switch (sensor_type) {
+        case SensorBase::SensorType::Barometer:
+            sensor_setting = std::unique_ptr<SensorSetting>(new BarometerSetting());
+            break;
+        case SensorBase::SensorType::Imu:
+            sensor_setting = std::unique_ptr<SensorSetting>(new ImuSetting());
+            break;
+        case SensorBase::SensorType::Gps:
+            sensor_setting = std::unique_ptr<SensorSetting>(new GpsSetting());
+            break;
+        case SensorBase::SensorType::Magnetometer:
+            sensor_setting = std::unique_ptr<SensorSetting>(new MagnetometerSetting());
+            break;
+        case SensorBase::SensorType::Distance:
+            sensor_setting = std::unique_ptr<SensorSetting>(new DistanceSetting());
+            break;
+        case SensorBase::SensorType::Lidar:
+            sensor_setting = std::unique_ptr<SensorSetting>(new LidarSetting());
+            break;
+        default:
+            break;
+        }
+
+        sensor_setting->sensor_type = sensor_type;
+        sensor_setting->sensor_name = sensor_name;
+        sensor_setting->enabled = enabled;
+
+        return sensor_setting;
+    }
+
+    static void initializeSensorSetting(SensorSetting* sensor_setting, const Settings& settings_json)
+    {
+        sensor_setting->enabled = settings_json.getBool("Enabled", sensor_setting->enabled);
+
+        switch (sensor_setting->sensor_type) {
+        case SensorBase::SensorType::Barometer:
+            break;
+        case SensorBase::SensorType::Imu:
+            break;
+        case SensorBase::SensorType::Gps:
+            break;
+        case SensorBase::SensorType::Magnetometer:
+            break;
+        case SensorBase::SensorType::Distance:
+            break;
+        case SensorBase::SensorType::Lidar:
+            initializeLidarSetting(static_cast<LidarSetting*>(sensor_setting), settings_json);
+            break;
+        default:
+            break;
```

I think default here should be to raise exception for unknown sensor_type